layer.md

```diff
@@ -1,89 +1,201 @@
-# Creating an Image Filesystem Changeset
+# Image Filesystem Changeset
 
-An example of creating an Image Filesystem Changeset follows.
+This document describes how to serialize a filesystem and filesystem changes like removed files into a blob called a layer.
+One or more layers are ordered on top of each other to create a complete filesystem.
+This document will use a concrete example to illustrate how to create and consume these filesystem layers.
 
-An image root filesystem is first created as an empty directory.
-Here is the initial empty directory structure for a changeset using the randomly-generated directory name `c3167915dc9d` ([actual layer DiffIDs are generated based on the content](#id_desc)).
+## Distributable Format
+
+Layer Changesets for the [mediatype](./media-types.md) `application/vnd.oci.image.layer.tar+gzip` MUST be packaged in [tar archive][tar-archive].
+Layer Changesets for the [mediatype](./media-types.md) `application/vnd.oci.image.layer.tar+gzip` MUST NOT include duplicate entries for file paths in the resulting [tar archive][tar-archive].
+
+## Change Types
+
+Types of changes that can occur in a changeset are:
+
+* Additions
+* Modifications
+* Removals
+
+Additions and Modifications are represented the same in the changeset tar archive.
+
+Removals are represented using "[whiteout](#whiteouts)" file entries (See [Representing Changes](#representing-changes)).
+
+### File Types
+
+Throughout this document section, the use of word "files" includes:
+
+* regular files
+* directories
+* sockets
+* symbolic links
+* block devices
+* character devices
+* FIFOs
+
+### File Attributes
+
+File attributes that are provided for Additions and Modifications include:
```

Comment:
On Wed, Sep 07, 2016 at 06:32:48PM -0700, Vincent Batts wrote:
> > +File attributes that are provided for Additions and Modifications include:
> 
> @wking your chain of links and footnotes _kill_ me.

Sorry, it's hard to preserve the continuity of a sub-thread when
GitHub keeps hiding it on us :p.  Links and footnotes are the best I
can do to make the older conversation discoverable without rehashing
all of it.

> The options your proposing are getting in the weeds. The requirement
> are known and have been worked with up to this point. If there are
> features needed to be added to tools, I'm certain folks will do what
> they need to do.

That sounds like (a) (Punt the specifics to users) from [1], and I'm
ok with that.  However, in that case, I see no particular use to call
out specific file types or attributes.  If a user wants to put some
oddball filetype/attribute into the tarball, more power to them.  If
an implementer wants to only support regular files, directories, and
mode (ignoring or erroring on symlinks, owner, mtime, etc.), more
power to them as well.  Sitting in the middle with a half-done spec
(e.g. requiring ‘xattrs’ but not defining it) doesn't seem helpful.

[1]: https://github.com/opencontainers/image-spec/pull/255#discussion_r77867267